lib/diagnostic.js

```diff
@@ -36,7 +58,11 @@ const newArray = function newArray(length, defaultValue) {
 // The function you write should return
 // the object passed in as an argument or the object created
 const addMethod = function addMethod(propertyName, method, obj) {
-
+  if(typeof obj[propertyName] === 'undefined') {
+    obj[propertyName] = method();
+    return true;
+  }
+  return false;
```

Comment:
This is not the correct behavior. If `obj` is undefined, the function should return a new object with a single property; that property's name is the value of `propertyName`, and its value is the method that was passed in.